lib/src/util/lazy.dart

```diff
@@ -0,0 +1,128 @@
+///
+/// Returns a `UiFactory` you can use just render in your tree. While the code for the lazy component is still loading,
+/// attempting to render it will suspend. Use <Suspense> to display a loading indicator while it’s loading.
+/// [load] is a function that should return a `Future<UiFactory<TProps>>` that resolves to the component to be rendered.
+/// [_config] should be a `UiFactoryConfig<TProps>` or `null` and is only `dynamic` to avoid an unnecessary cast in the boilerplate.
+///
+/// React will not call [load] until the first time the component is rendered.
+/// After React first calls [load], it will wait for it to resolve, and then render the resolved value.
+/// Both the returned Future and the Future's resolved value will be cached, so React will not call [load] more than once.
+/// If the Future rejects, React will throw the rejection reason for the nearest Error Boundary to handle.
+///
+/// Example:
+/// ```dart
+/// import 'package:over_react/over_react.dart';
+///
+/// part 'main.over_react.g.dart';
+///
+/// mixin ALazyComponentPropsMixin on UiProps {}
+///
+/// UiFactory<ALazyComponentPropsMixin> ALazyComponent = lazy(
+///   () async {
+///     final componentModule = await loadComponent();
+///     return uiForwardRef(
+///       (props, ref) {
+///         return (componentModule.AnotherComponent()
+///           ..ref = ref
+///           ..addProps(props)
+///         )(props.children);
+///       },
+///       _$ALazyComponentConfig,
+///     );
+///   },
+///   _$ALazyComponentConfig
+/// );
+/// ```
+///
+/// > __NOTE:__ A lazy component MUST be wrapped with a `Suspense` component to provide a fallback ui while it loads.
+///
+/// ```dart
+/// (Suspense()
+///   ..fallback = Dom.p()('Loading...')
+/// )(
+///    ALazyComponent()(),
+/// );
+/// ```
+/// See: <https://react.dev/reference/react/lazy>.
+UiFactory<TProps> lazy<TProps extends UiProps>(
+    Future<UiFactory<TProps>> Function() load, /* UiFactoryConfig<TProps> */ dynamic _config) {
+      _config ??= UiFactoryConfig();
+  if (_config is! UiFactoryConfig<TProps>) {
+    throw ArgumentError('_config is required when using a custom props class and should be a UiFactoryConfig<TProps>. Make sure you are '
+        r'using either the generated factory config (i.e. _$FooConfig) or manually declaring your config correctly.');
+  }
+  // ignore: invalid_use_of_protected_member
+  var propsFactory = _config.propsFactory;
+
+  final lazyFactoryProxy = react.lazy(() async {
+    final factory = await load();
+    // By using a wrapper uiForwardRef it ensures that we have a matching factory proxy type given to react-dart's lazy,
+    // a `ReactDartWrappedComponentFactoryProxy`. This is necessary to have consistent prop conversions since we don't
+    // have access to the original factory proxy outside of this async block.
+    final wrapper = uiForwardRef<TProps>(
```

Comment:
See note [here](https://github.com/Workiva/react-dart/pull/406#discussion_r1779289577) about moving the wrapper into react-dart

Reply:
So, i tried removing the wrapper at this level and it caused test failures and wasn't converting props like function props in some of the type assertion tests i added... seems like we need it at both levels. :/

Reply:
Hmm interesting. To me it doesn't seem like we'd need it at both levels, unless it's some JS interop tradeoff case that we might not need to support.

Once I get a chance I can try removing it locally and dig into what the specific failures are

Reply:
Hm I removed the wrapper and am not seeing any test failures, locally or in CI; here's a PR: https://github.com/Workiva/over_react/pull/953

Oh, is it possible that you were trying with an outdated version of react-dart, from before the wrapper was removed? You can force it to get the latest with `dart pub upgrade react`

Reply:
>Oh, is it possible that you were trying with an outdated version of react-dart, from before the wrapper was removed? You can force it to get the latest with dart pub upgrade react

Crap yes that probably was it... damn.....